### PR TITLE
 RHICOMPL-181 - Delete hosts when notified by the inventory 

### PR DESCRIPTION
**Pull request by dLobatog**
https://github.com/RedHatInsights/s2i-insights-compliance/pull/12

### CODE REVIEW

File: app/consumers/inventory_events_consumer.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    return unless msg_value['type'] == 'delete'
+
+    DeleteHost.perform_async(msg_value['id'])
+  end
+
+  private
+
+  def logger
+    Rails.logger
```
Comment (akofink) [inline app/consumers/inventory_events_consumer.rb]: Why rails logger and not Sidekiq?

File: app/jobs/delete_host.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+# Job meant to delete hosts and associated objects asynchronously
+class DeleteHost
+  include Sidekiq::Worker
+
+  def perform(host_id)
+    RuleResult.where(host_id: host_id).delete_all
+    Host.find(host_id).destroy
+  end
```
Comment (akofink) [inline app/jobs/delete_host.rb]: Let's wrap this in a transaction

File: app/consumers/inventory_events_consumer.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class InventoryEventsConsumer < ApplicationConsumer
+  subscribes_to Settings.platform_kafka_inventory_topic
+
+  def process(message)
+    msg_value = JSON.parse(message.value)
+    logger.info "Received message, enqueueing: #{message.value}"
+
+    return unless msg_value['type'] == 'delete'
+
+    DeleteHost.perform_async(msg_value['id'])
```
Comment (akofink) [inline app/consumers/inventory_events_consumer.rb]: I'd much prefer passing the entire message for the logs (on potential failure). It'd also keep us more flexible for future changes (like if we one day support deletion of multiple hosts).

File: app/consumers/inventory_events_consumer.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Receives messages from the Kafka topic, converts them into jobs
+# for processing
+class InventoryEventsConsumer < ApplicationConsumer
+  subscribes_to Settings.platform_kafka_inventory_topic
```
Comment (akofink) [inline app/consumers/inventory_events_consumer.rb]: Missing the config file update?

File: app/consumers/inventory_events_consumer.rb (hunk 1 above)
Comment (dLobatog) [reply app/consumers/inventory_events_consumer.rb]: The `consumer` runs using Racecar, not sidekiq

File: app/jobs/delete_host.rb (hunk 2 above)
Comment (dLobatog) [reply app/jobs/delete_host.rb]: :+1:

File: app/consumers/inventory_events_consumer.rb (hunk 3 above)
Comment (dLobatog) [reply app/consumers/inventory_events_consumer.rb]: :+1:

File: app/consumers/inventory_events_consumer.rb (hunk 4 above)
Comment (dLobatog) [reply app/consumers/inventory_events_consumer.rb]: Definitely :sob:

**Pull request merged**